src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/ISearchParameterExpressionParser.cs

```diff
@@ -13,6 +13,7 @@ public interface ISearchParameterExpressionParser
         Expression Parse(
             SearchParameterInfo searchParameter,
             SearchModifierCode? modifier,
+            string targetTypeModifier,
             string value);
```

Comment:
nit: Since you touching this method....
I would consider to change order of these parameters and provide default values for optional parameters.
Make necessary parameters first, and put all optional parameters in the back.

Code like that: `Parse(searchParameter, null, null, value)` looks much better when it's just `Parse(searchParameter, value);`

It's a bit of extra refactoring, but i think it code would look cleaner and it's worth it.

Reply:
After adding `SearchModifier` class in f4409b5 the `modifier` is only used as an optional parameter in a few unit tests.
I think, I prefer them in the current order (which is the order they come in Http query).